cmd/core.go

```diff
@@ -0,0 +1,37 @@
+package cmd
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var coreCmd = &cobra.Command{
+	Use:     "core",
+	Aliases: []string{"homeassistant", "home-assistant", "ha"},
+	Short:   "Provides control of the Home Assistant Core",
+	Long: `
+This command provides a set of subcommands to control the Home Assistant Core
+instance running on this installation.
+
+It provides commands to control Home Assistant Core (start, stop, restart),
+but also allows you to check your Home Assistant Core configuration.
+Furthermore, some options can be set and allows for upgrading/downgrading
+Home Assistant Core.
+`,
+	Example: `
+  ha core check
+  ha core restart
+  ha core update
+	ha core update --version 0.97.2`,
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		if os.Args[1] != "core" {
+			cmd.PrintErrf("The use of '%s' is deprecated, please use 'core' instead!\n", os.Args[1])
```

Comment:
Didn't have time to test this,
but does this work, or give a deprecation warning ?
`ha -d core check`

Reply:
That will most likely give a warning, however, while Cobra provides a method to ask for the called alias, it doesn't work for parent commands.

So this is not ideal but will do for 99% of the cases (manual command line usage). I do not expect this to be around for a long time.

Reply:
Improved the handling (still not pretty, but better).
Since `homeassistant`, `ha` and `hassos` aren't used anywhere anymore, we can safely check if either of the arguments contains just that.

So I added a small loop over all arguments to search for it.